src/Bootstrappers/PrefixCacheTenancyBootstrapper.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Closure;
+use Illuminate\Cache\CacheManager;
+use Illuminate\Cache\Repository;
+use Illuminate\Contracts\Config\Repository as ConfigRepository;
+use Illuminate\Support\Facades\Cache;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class PrefixCacheTenancyBootstrapper implements TenancyBootstrapper
+{
+    protected string|null $originalPrefix = null;
+    public static array $tenantCacheStores = []; // E.g. ['redis']
+    public static Closure|null $prefixGenerator = null;
+
+    public function __construct(
+        protected ConfigRepository $config,
+        protected CacheManager $cacheManager,
+    ) {
+    }
+
+    public function bootstrap(Tenant $tenant): void
+    {
+        $this->originalPrefix = $this->config->get('cache.prefix');
+
+        // Use default prefix generator if the prefix generator isn't set
+        static::$prefixGenerator ??= function (Tenant $tenant) {
+            return $this->originalPrefix . $this->config->get('tenancy.cache.prefix_base') . $tenant->getTenantKey();
+        };
```

Comment:
Was that in this PR or elsewhere that I wrote that it might be cleaner to keep the variable always null if it's not user-defined, and using the default logic when that property is null? Can't find it now but I think I mentioned this in some context.

Reply:
If we made that change, we'd also remove any `static::$prefixGenerator = null` calls to avoid overriding the user-defined prefix generator. In fact, it seems like the code already overrides it. Initialize -> revert -> initialize and it will be gone (due to line 53).

Reply:
Also: this in the `PrefixCacheBootstrapperTest` would have to be updated:
```php
$prefixGenerator = PrefixCacheTenancyBootstrapper::$prefixGenerator;

cache()->set('key', 'tenantone-value');

$expectPrefixToBe($prefixGenerator($tenant1));
```

Probably add a public method to the bootstrapper for generating prefixes 👍🏻

Reply:
Or you could manually define the generator in the test, but adding a public method is simpler and it can be a helpful thing to be available for people to use. Can assist with debugging, for example.

Reply:
> Initialize -> revert -> initialize and it will be gone (due to line 53).

You're right, I didn't think about that. I'll fix this

Reply:
There's now a public `generatePrefix()` method, and the defaulting is inline